packages/host/app/services/store.ts

```diff
@@ -231,7 +248,32 @@ export default class StoreService extends Service {
     if (cached) {
       return cached as T;
     }
-    return await this.getCard<T>(url, undefined);
+    return await this.getCard<T>({ urlOrDoc: url });
+  }
+
+  getAutoSaveState(card: CardDef): AutoSaveState | undefined {
+    return this.autoSaveStates.get(card);
+  }
+
+  private async guardAgainstUnknownInstances(instance: CardDef) {
+    if (instance.id && this.identityContext.get(instance.id) !== instance) {
+      throw new Error(
+        `tried to add ${instance.id} to the store, but the store already has a different instance with this id. Please obtain the instances that already exists from the store`,
+      );
+    }
+    let api = await this.cardService.getAPI();
+    let deps = getDeps(api, instance);
+    for (let dep of deps) {
+      if (dep.id && this.identityContext.get(dep.id) !== dep) {
+        throw new Error(
+          `encountered a dependency, ${dep.id}, of ${instance.id} when adding ${instance.id} to the store, but the store ${
+            this.identityContext.get(dep.id)
+              ? "already has a different instance with this dependency's id"
+              : "has no entry for this dependency's id"
+          }. Please obtain the instances that already exists from the store`,
```

Comment:
in the case where the dep just doesn't exist yet in the identity map we should just add it instead of throwing.